src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -14,60 +13,32 @@
-    // are much more efficient when the SocketAsyncEventArg instances can be reused; as such,
-    // at present we use them for ReceiveAsync and SendAsync, caching an instance for each.
-    // In the future we could potentially maintain a global cache of instances used for accepts
-    // and connects, and potentially separate per-socket instances for Receive{Message}FromAsync,
-    // which would need different instances from ReceiveAsync due to having different results
-    // and thus different Completed logic.  We also currently fall back to APM implementations
-    // when the single cached instance for each of send/receive is otherwise in use; we could
-    // potentially also employ a global pool from which to pull in such situations.
-
     public partial class Socket
     {
-        /// <summary>Handler for completed AcceptAsync operations.</summary>
-        private static readonly EventHandler<SocketAsyncEventArgs> AcceptCompletedHandler = (s, e) => CompleteAccept((Socket)s!, (TaskSocketAsyncEventArgs<Socket>)e);
-        /// <summary>Handler for completed ReceiveAsync operations.</summary>
-        private static readonly EventHandler<SocketAsyncEventArgs> ReceiveCompletedHandler = (s, e) => CompleteSendReceive((Socket)s!, (Int32TaskSocketAsyncEventArgs)e, isReceive: true);
-        /// <summary>Handler for completed SendAsync operations.</summary>
-        private static readonly EventHandler<SocketAsyncEventArgs> SendCompletedHandler = (s, e) => CompleteSendReceive((Socket)s!, (Int32TaskSocketAsyncEventArgs)e, isReceive: false);
-        /// <summary>
-        /// Sentinel that can be stored into one of the Socket cached fields to indicate that an instance
-        /// was previously created but is currently being used by another concurrent operation.
-        /// </summary>
-        private static readonly TaskSocketAsyncEventArgs<Socket> s_rentedSocketSentinel = new TaskSocketAsyncEventArgs<Socket>();
-        /// <summary>
-        /// Sentinel that can be stored into one of the Int32 fields to indicate that an instance
-        /// was previously created but is currently being used by another concurrent operation.
-        /// </summary>
-        private static readonly Int32TaskSocketAsyncEventArgs s_rentedInt32Sentinel = new Int32TaskSocketAsyncEventArgs();
         /// <summary>Cached task with a 0 value.</summary>
         private static readonly Task<int> s_zeroTask = Task.FromResult(0);
 
-        /// <summary>Cached event args used with Task-based async operations.</summary>
-        private CachedEventArgs? _cachedTaskEventArgs;
+        /// <summary>Cached instance for accept operations.</summary>
+        private TaskSocketAsyncEventArgs<Socket>? _acceptEventArgs;
+
+        /// <summary>Cached instance for receive operations that return <see cref="ValueTask{Int32}"/>. Also used for ConnectAsync operations.</summary>
+        private AwaitableSocketAsyncEventArgs? _singleBufferReceiveEventArgs;
```

Comment:
Just an observation. I usually name these with the word cached in their name, `_cachedBufferReceiveEventArgs` and find that this helps distinguish working code from cache management code.

Reply:
Thanks.